### PR TITLE
Auto: implement issue #675

### LINKED ISSUE #675: Add import/export ORM models and migrations
Subtask of #34

Define ORM models for ImportJob and ExportJob in src/db/models/. ImportJob stores: id, tenant_id, entity_type, file_path, status (pending/running/done/failed), total_rows, processed_rows, error_rows, created_at. ExportJob stores: id, tenant_id, entity_type, fields (JSON), filters (JSON), file_path, status, expires_at. Add them to alembic/env.py imports, then generate a migration. Do not implement any service or router logic yet.

Dev-Plan: docs/dev-plan/00-foundations/0675-add-import-export-orm-models-and-migrations.md

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #675

Plan: `.plans/issue-675.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-675.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Summary by CodeRabbit

* **New Features**
  * Introduced import and export job tracking functionality to monitor the status, progress, and metadata of bulk data operations, including row counts, file paths, and expiration settings.

* **Tests**
  * Added comprehensive integration tests verifying job creation, persistence, retrieval, and serialization across multiple scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/709?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: .plans/issue-675.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- `src/db/models/import_job.py` — new file: `ImportJobModel` with id, tenant_id, entity_type, file_path, status, total_rows, processed_rows, error_rows, created_at, updated_at
+- `src/db/models/export_job.py` — new file: `ExportJobModel` with id, tenant_id, entity_type, fields (JSON), filters (JSON), file_path, status, expires_at, created_at, updated_at
+- `alembic/versions/<hash>_add_import_export_jobs.py` — new migration file creating both tables, keyed to head revision `9d8e7f6a5b3c`
+
+## Implementation Steps
+
+1. **Create `src/db/models/import_job.py`** following the existing pattern (e.g. `task.py`, `workflow.py`):
+   - `__tablename__ = "import_jobs"`
+   - Fields: `id` (Integer PK), `tenant_id` (Integer, indexed), `entity_type` (String(100)), `file_path` (String(1000)), `status` (String(50), default `"pending"`), `total_rows` (Integer, default 0), `processed_rows` (Integer, default 0), `error_rows` (Integer, default 0), `created_at` and `updated_at` (DateTime with `server_default=func.now()`)
+   - `to_dict()` method serialising all fields with isoformat for datetimes
```
Comment (yihua-zhuo) [inline .plans/issue-675.md:20]: **INFO** (suggestion)

**Issue:** Plan lists fields using server_default defaults, but line 24 field spec (export_jobs) does not mention onupdate for updated_at — the plan does not specify whether export_jobs.updated_at should auto-update.

**Suggestion:** Clarify whether export_jobs.updated_at should use onupdate=func.now() to match import_jobs behavior, or document why it does not need it.

File: alembic/versions/a52e1317da90_add_import_export_jobs.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = 'a52e1317da90'
+down_revision: str | None = '9d8e7f6a5b3c'
+branch_labels: Sequence[str] | None = None
+depends_on: str | None = None
```
Comment (yihua-zhuo) [inline alembic/versions/a52e1317da90_add_import_export_jobs.py:19]: **WARNING** (suggestion)

**Issue:** depends_on is None but plan (line 13) specifies depends_on: 9d8e7f6a5b3c. While down_revision is correctly set to 9d8e7f6a5b3c, an explicit depends_on is the documented contract.

**Suggestion:** Set depends_on: str | None = '9d8e7f6a5b3c' on line 19 to match the plan's contract.

File: src/db/models/export_job.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    tenant_id: Mapped[int] = mapped_column(Integer, nullable=False, index=True)
+    entity_type: Mapped[str] = mapped_column(String(100), nullable=False)
+    fields: Mapped[dict] = mapped_column(JSONB, default=dict, nullable=False)
+    filters: Mapped[dict] = mapped_column(JSONB, default=dict, nullable=False)
+    file_path: Mapped[str] = mapped_column(String(1000), nullable=False)
+    status: Mapped[str] = mapped_column(String(50), default="pending", nullable=False)
+    expires_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
+    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
+    updated_at: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False
```
Comment (yihua-zhuo) [inline src/db/models/export_job.py:27]: **WARNING** (suggestion)

**Issue:** Rule 139: export_jobs.updated_at has server_default but no onupdate, while import_job.updated_at has both server_default and onupdate. This inconsistency means export job row updates will not auto-update the updated_at timestamp.

**Suggestion:** Add onupdate=func.now() to the updated_at mapped_column: mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

File: tests/integration/test_import_export_models_integration.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        async_session.add(
+            ImportJobModel(
+                tenant_id=9999,
+                entity_type="customer",
+                file_path="/uploads/other_tenant.csv",
+            )
+        )
+        await async_session.flush()
+
+        result = await async_session.execute(
```
Comment (yihua-zhuo) [inline tests/integration/test_import_export_models_integration.py:98]: **WARNING** (suggestion)

**Issue:** test_query_by_tenant_id for ImportJobModel does not verify cross-tenant isolation — it never queries tenant 9999 to confirm those rows are not returned, so a missing WHERE clause would pass silently (Rule 126).

**Suggestion:** Add an assertion after the existing query: result_other = await async_session.execute(select(ImportJobModel).where(ImportJobModel.tenant_id == 9999)); other_rows = result_other.scalars().all(); assert len(other_rows) == 1

File: tests/integration/test_import_export_models_integration.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                tenant_id=9999,
+                entity_type="customer",
+                fields={},
+                filters={},
+                file_path="/exports/other_tenant.csv",
+            )
+        )
+        await async_session.flush()
+
+        result = await async_session.execute(
```
Comment (yihua-zhuo) [inline tests/integration/test_import_export_models_integration.py:186]: **WARNING** (suggestion)

**Issue:** test_query_by_tenant_id for ExportJobModel does not verify cross-tenant isolation — it never queries tenant 9999 to confirm those rows are not returned (Rule 126).

**Suggestion:** Add: result_other = await async_session.execute(select(ExportJobModel).where(ExportJobModel.tenant_id == 9999)); other_rows = result_other.scalars().all(); assert len(other_rows) == 1

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 5

**Summary:** Reviewed 5 file(s) between origin/master and origin/auto/issue-675.

### Issues without a usable file/line
- `src/db/models/import_job.py`: import_job.py model follows the established codebase pattern correctly: all fields match the plan, server_default + onupdate are both present on updated_at, to_dict() serialization is correct, and pkgutil auto-discovery via db.models.__init__.py will register it without changes.
- `tests/integration/test_import_export_models_integration.py`: No failure/edge-case tests exist for ImportJobModel or ExportJobModel — missing tests for invalid status values, negative row counts, or missing required fields (Rule 65).
- `tests/integration/test_import_export_models_integration.py`: No test verifies that updated_at auto-updates on an UPDATE operation. The migration uses onupdate=func.now() but this behavior is never exercised in tests (Rule 65).
